### PR TITLE
Sanitize custom Plasma theme name

### PR DESCRIPTION
**Pull request by DarNCelsius**
Plasma prefixes the theme name in `kwinrc` with `"__aurorae__svg__"` for custom themes it seems.

The name shows up as normal in settings:
plasmatheme
Theme desktop entry:
```
[Desktop Entry]
Name=Nord-Aurorae
Comment=Nord-Aurorae
X-KDE-PluginInfo-Author=l4k1
X-KDE-PluginInfo-Email=l4k1987@gmail.com
X-KDE-PluginInfo-Version=1.0
X-KDE-PluginInfo-Name=Nord-Aurorae
X-KDE-PluginInfo-Category=
X-KDE-PluginInfo-Depends=
X-KDE-PluginInfo-License=GPLv3
X-KDE-PluginInfo-EnabledByDefault=true
```
The only place the string appears is in the `kwinrc` file, so I think #67 is valid after all.

### CODE REVIEW

**Comment by LinusDierheimer** [comment]
Looking at the neofetch source code, there actually is more that needs to be removed:
```awk
gsub(/theme=.*qml_|theme=.*svg__/,"",$0);
```
Can you implement that too?

**Comment by DarNCelsius** [comment]
I'll have a look.

The second pattern is matching the same as this I guess. The Plastik theme is old and has `"kwin4_decoration_qml_"` prefixed, so would probably fit the first.

This is Neofetch with Plastik:

**Comment by LinusDierheimer** [comment]
".*" is a regex expression to match a string of any length

**Comment by DarNCelsius** [comment]
Having a hard time figuring out the ffsubstring routines.

With this string: `kwin4_decoration_qml_plastik` in "theme", calling
```c
idx = ffStrbufFirstIndexAfterS(&theme, 0, "qml_");
```
and then
```c
ffStrbufSubstrAfter(&theme, idx);
```
Should produce what?

**Comment by LinusDierheimer** [comment]
`qml_plastik` i guess. Also you can use `ffStrbufFirstIndexS` instead of using start index 0

**Comment by DarNCelsius** [comment]
It returns `"in4_decoration_qml_plastik"`

**Comment by LinusDierheimer** [comment]
Will look into this.

**Comment by DarNCelsius** [comment]
Shouldn't it produce "plastik" though? Since it's the index _after_?

**Comment by LinusDierheimer** [comment]
No, the after is means it starts searching after a given index. Currently doing a rewrite of those functions with better naming.

**Comment by DarNCelsius** [comment]
I see. Is there currently a substring function that would produce the string after (or before) the string you search for?

**Comment by LinusDierheimer** [comment]
There currently isn't, the combination of the two methods is the best method currently. If you feel like it is needed, you can write a `ffStrbufRemoveStartS(const FFstrbuf* strbuf, const char* str)` or similar. 
I pushed a commit that renames some strbuf methods, the method u asked about is now called `ffStrbufNextIndexS`. Hope this makes things clearer.

**Comment by DarNCelsius** [comment]
The combination of the two isn't producing that result though...

**Comment by LinusDierheimer** [comment]
This works:
```c
FFstrbuf theme;
ffStrbufInit(&theme);
ffStrbufAppendS(&theme, "kwin4_decoration_qml_plastik");

uint32_t index = ffStrbufFirstIndexS(&theme, "qml_");
ffStrbufSubstrAfter(&theme, index + 3); //index of q plus ml_
ffStrbufPutTo(&theme, stdout);
```
Prints `plastik`.

**Comment by DarNCelsius** [comment]
So yes then. :+1:

**Comment by DarNCelsius** [comment]
New version works with both and no matches

File: src/modules/wmtheme.c [hunk 1]
```diff
@@ -45,6 +45,17 @@ static void printWMThemeFromConfigFile(FFinstance* instance, const char* configF
         return;
     }
 
+    // Remove Plasma-generated prefixes
+    uint32_t idx = 0;
+
+    idx = ffStrbufFirstIndexS(&theme, "qml_");
+    if((idx != theme.length))
```
Comment (LinusDierheimer) [inline src/modules/wmtheme.c:52]: Why the double brackets?

Comment (DarNCelsius) [reply src/modules/wmtheme.c:52]: Oops. Copy paste error. I was going for a one liner prior. Something like
```c
if((idx = ffStrbufFirstIndexS(&theme, "qml_") != theme.length))
```
Couldn't get it to work.

Comment (LinusDierheimer) [reply src/modules/wmtheme.c]: Probably because the index methods remove the index of the terminating null byte if the search failed.

**Comment by LinusDierheimer** [comment]
Thank you.

**Pull request merged**